lib/casmutils/xtal/structure.cxx

```diff
@@ -14,28 +14,59 @@ namespace extend
 
 namespace rewrap
 {
+
+template <> void Structure::_update_using<CASM::xtal::BasicStructure>(const CASM::xtal::BasicStructure& struc)
+{
+    _update_simple_from_basic();
+    _update_internals_from_basic();
+}
+
+template <> void Structure::_update_using<CASM::xtal::SimpleStructure>(const CASM::xtal::SimpleStructure& struc)
+{
+    _update_basic_from_simple();
+    _update_internals_from_simple();
+}
+
+template <> void Structure::_update_using<casmutils::xtal::Structure>(const casmutils::xtal::Structure& struc)
+{
+    _update_basic_from_internals();
+    _update_simple_from_internals();
+}
+
+/// Return *this as a CASM::BasicStructure
+template <> const CASM::xtal::SimpleStructure& Structure::__get<CASM::xtal::SimpleStructure>() const
+{
+    return this->casm_simplestructure;
+}
+
+/// Return *this as a CASM::BasicStructure
+template <> const CASM::xtal::BasicStructure& Structure::__get<CASM::xtal::BasicStructure>() const
+{
+    return this->casm_basicstructure;
+}
+
 Structure::Structure(const CASM::xtal::BasicStructure& init_struc)
     : structure_lattice(init_struc.lattice()), casm_basicstructure(init_struc)
 {
-    _update_using("basic");
+    _update_using(this->__get<CASM::xtal::BasicStructure>());
```

Comment:
If you use template specialization you don't need to pass any arguments:
```
this->_update_using<CASM::xtal::BasicStructure>();
```
Your implementation is discarding the passed structure anyway.